### PR TITLE
Added entry for Best-First Search

### PR DESCRIPTION
**Pull request by HishamT**
### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/ai/concepts/search algorithms/best first search.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+In a weighted graph, the total cost to go from the start node to the goal node is the sum of the cost of every visted edge along the path. The values from the evaluation function do not contribute to the total cost. They are only there to guide the search.
+
+## Best-First Search Algorithms
+The following is a list of Best-First search algorithms.
+
+- Greedy Best-First Search
```
Comment (SSwiniarski) [inline content/ai/concepts/search algorithms/best first search.md:26]: Can we have a one line summary for each of these?

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@HishamT I dis a first review and only had one comment.

File: content/ai/concepts/search algorithms/best first search.md (hunk 1 above)
Comment (HishamT) [reply content/ai/concepts/search algorithms/best first search.md]: Done! :grin:

**Review by SSwiniarski — APPROVED** [review]
@HishamT Looks good for another review.

File: content/ai/concepts/search algorithms/best first search.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+In a weighted graph, the total cost from the start node to the goal node is the sum of the cost of every visited edge along the path. The values from the evaluation function do not contribute to the total cost. They are only there to guide the search.
+
+## Best-First Search Algorithms
```
Comment (caupolicandiaz) [inline content/ai/concepts/search algorithms/best first search.md:23]:
```suggestion
## Best-First Search Algorithms

```

File: content/ai/concepts/search algorithms/best first search.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  - 'AI'
+Tags:
+  - 'Search Algorithms'
+  - 'AI'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/machine-learning-ai-engineering-foundations'
+---
+
+**Best-First Search** is a class of search algorithms that searches a graph (the graph can be weighted or unweighted) for a path from some start node to a goal node. These algorithms are known as informed search algorithms, meaning that they incorporate information regarding the location of the goal node relative to any other node. This information is quantified by an evaluation function denoted by _f(x)_. In a weighted graph, the evaluation function can combine information regarding the edge weights and some heuristic function, _h(x)_, that represents a measure from a node (_x_ in this case) to the goal node. When searching for the goal node, a BFS algorithm (not to be confused with Breadth-First Search) selects the node with the most promising value determined by the evaluation function.
```
Comment (caupolicandiaz) [inline content/ai/concepts/search algorithms/best first search.md:15]:
```suggestion
**Best-first search** is a class of search algorithms that explores a graph (the graph can be weighted or unweighted) for a path from some start node to a goal node. These algorithms are known as informed search algorithms, meaning that they incorporate information regarding the location of the goal node relative to any other node. This information is quantified by an evaluation function denoted by _f(x)_. In a weighted graph, the evaluation function can combine information regarding the edge weights and some heuristic function, _h(x)_, that represents a measure from a node (_x_ in this case) to the goal node. When searching for the goal node, a BFS algorithm (not to be confused with breadth-first search) selects the node with the most promising value determined by the evaluation function.
```

File: content/ai/concepts/search algorithms/best first search.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+---
+Title: 'Best-First Search'
+Description: 'A graph searching algorithm that expands a node based on an evaluation function.'
```
Comment (caupolicandiaz) [inline content/ai/concepts/search algorithms/best first search.md:3]:
```suggestion
Description: 'A graph searching algorithm that determines a path between nodes based on an evaluation function.'
```

Comment (caupolicandiaz) [reply content/ai/concepts/search algorithms/best first search.md:3]: I just found "expands a node" very abstract and opaque. I've provided an alternative, but feel free to try something else.

File: content/ai/concepts/search algorithms/best first search.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+In a weighted graph, the total cost from the start node to the goal node is the sum of the cost of every visited edge along the path. The values from the evaluation function do not contribute to the total cost. They are only there to guide the search.
+
+## Best-First Search Algorithms
+The following is a list of Best-First search algorithms.
+
+- Greedy Best-First Search: A search algorithms where the evaluation function is strictly equal to the heuristic function, disregarding the edge weights in a weighted graph.
```
Comment (caupolicandiaz) [inline content/ai/concepts/search algorithms/best first search.md:26]:
```suggestion
- Greedy Best-First Search: A search algorithm where the evaluation function is strictly equal to the heuristic function, disregarding the edge weights in a weighted graph.
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@HishamT Overall, looks good. Just a few comments, mostly language suggestions. Additionally, we are going to move forward with a terms folder, so, please add a folder and relocate this entry as part of this PR. Thanks

**Comment by HishamT** [comment]
@SSwiniarski, @caupolicandiaz, thank you both so much for your reviews. I have implemented your suggestions. I will move this to a new folder as you suggested, Paul.

**Comment by caupolicandiaz** [comment]
> @SSwiniarski, @caupolicandiaz, thank you both so much for your reviews. I have implemented your suggestions. I will move this to a new folder as you suggested, Paul.

Apologies Hisham, I don't think I was clear in my comment. We are going to repurpose the search-algorithms entry that was recently created as the landing document for search within AI. The specific algos will live in a terms folder within search, it should be- content/ai/concepts/search-algorithms/terms/best-first-search/best-first-search.md.

Let me know if you have any questions. Thanks

**Review by caupolicandiaz — APPROVED** [review]
@HishamT Apologies again for all the back n forth. I've gone ahead and moved the search-algorithms doc into this PR so everything is all together.

**Comment by yangc95** [comment]
🤖👏 @HishamT here's the live link: https://www.codecademy.com/resources/docs/ai/search-algorithms/best-first-search

**Pull request merged**